ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -459,13 +459,21 @@ public Optional<TransactionSimulatorResult> processWithWorldUpdater(
   }
 
   @VisibleForTesting
-  protected void applyOverrides(final MutableAccount account, final StateOverride override) {
+  protected static void applyOverrides(final MutableAccount account, final StateOverride override) {
     LOG.debug("applying overrides to state for account {}", account.getAddress());
     override.getNonce().ifPresent(account::setNonce);
-    if (override.getBalance().isPresent()) {
-      account.setBalance(override.getBalance().get());
-    }
-    override.getCode().ifPresent(n -> account.setCode(Bytes.fromHexString(n)));
+    override.getBalance().ifPresent(account::setBalance);
+    override.getCode().ifPresent(code -> account.setCode(Bytes.fromHexString(code)));
+    override
+        .getState()
+        .ifPresent(
+            d -> {
+              account.clearStorage();
+              d.forEach(
+                  (key, value) ->
+                      account.setStorageValue(
+                          UInt256.fromHexString(key), UInt256.fromHexString(value)));
+            });
```

Comment:
maybe extract lines 472-475 to a method and call it for state and stateDiff - something like setStorageValues or applyStorageOverrides ...

Reply:
or maybe 470-475 and pass a boolean for whether to clear first